packages/app-bridge/src/react/useDocumentPages.ts

```diff
@@ -12,17 +12,38 @@ export type DocumentPageEvent = {
     documentPage: DocumentPage | { id: number };
 };
 
-const sortDocumentPages = (a: DocumentPage, b: DocumentPage) => (a.sort && b.sort ? a.sort - b.sort : 0);
-
-export const useDocumentPages = (appBridge: AppBridgeBlock | AppBridgeTheme, documentId: number) => {
+export const useDocumentPages = (
+    appBridge: AppBridgeBlock | AppBridgeTheme,
+    documentId?: number,
+    categoryId?: number,
+    allPages?: boolean,
+) => {
     const [pages, setPages] = useState<Map<number, DocumentPage>>(new Map([]));
+    const [categorizedPages, setCategorizedPages] = useState<Map<number, DocumentPage>>(new Map([]));
+    const [uncategorizedPages, setUncategorizedPages] = useState<Map<number, DocumentPage>>(new Map([]));
     const [isLoading, setIsLoading] = useState<boolean>(true);
 
     const refetch = useCallback(async () => {
         setIsLoading(true);
-        const pages = await fetchDocumentPages(appBridge, documentId);
 
-        setPages(pages);
+        if (categoryId) {
+            const categorizedPages = await fetchCategorizedDocumentPages(appBridge, categoryId);
+
+            setCategorizedPages(categorizedPages);
+        }
+
+        if (documentId) {
```

Comment:
this logic is not ok, it will do redundant calls for all pages. If all pages are set, it should only call fetchDocumentPages. Also, the boolean allPages, should be passed in as the config object.